### DISCUSSION TITLE
How do I download 10 m and 80 m wind data for a range of dates?

### DISCUSSION THREAD

**Discussion by blaylockbk** [blaylockbk/Herbie#9] [Q&A] [answered]
From an email:

"I want to download just the 10 m and 80 m winds for the model analysis for a short period of time."

**Comment by blaylockbk** [comment, marked as answer]
The secret sauce is knowing how to specify the search string to find both the U and V wind field for 10 m and 80 m winds. In this case, we use regular expression to set `searchString='(?:U|V)GRD:(?:10|80) m'`

```python
from herbie.tools import fast_Herbie_download
import pandas as pd

# Range of dates (in UTC) at hourly frequency, for 3 days
DATES = pd.date_range('2021-02-01 00:00', '2021-02-03 00:00', freq='1H')

# Download the data, just U and V winds at 10 and 80 m for the model analysis
d = fast_Herbie_download(
    DATES, 
    model='hrrr', 
    product='sfc',
    searchString='(?:U|V)GRD:(?:10|80) m', 
    fxx=0
)
```
> More examples of Fast Herbie in the [docs](https://blaylockbk.github.io/Herbie/_build/html/user_guide/notebooks/tutorial.html#Working-with-multiple-Herbie-objects)

> More examples of specifying the searchString in the [docs](https://blaylockbk.github.io/Herbie/_build/html/user_guide/searchString.html)

**Reply by blaylockbk** [reply]
Of course, if you only need data for a small area instead of the full CONUS domain, you can use the HRRR-Zarr archive on AWS. I haven't gone down this road yet, but there are example notebooks [linked here](https://blaylockbk.github.io/HRRR_archive_download/_build/html/user_guide/notebooks.html).

**Reply by jahanbani** [reply]
@blaylockbk 
The above answer throws an error. Would it be possible to download data for a range of time using the latest Herbie?

**Reply by blaylockbk** [reply]
Ah, yes. The function name changed since this post was published. The correct function is `fast_Herbie_download`

```python
from herbie.tools import fast_Herbie_download
```

I updated the post.
